### DISCUSSION TITLE
Undefined index languageDescription after uninstalling languages

### DISCUSSION THREAD

**Discussion by kanbeq** [datenstrom/community#667] [Report a bug] [answered]
Hi,
after uninstalling (via commands) the default installed languages (English and German), I get the following error:

Notice: Undefined index: languageDescription in ... \ yellow \ system \ extensions \ core.php on line 3544

When I install again both languages, the error disappears. When I install one of them, the error still shows.
The problem is, I want to get rid of these languages, and reinstalling them is not the solution.

I am asking for help, or for possible guidance on how I could fix this error.

**Comment by schulle4u** [comment]
I can't reproduce the problem on a fresh installation. This is what I did, assuming you are using polish language: 

1. Clone a fresh copy of `datenstrom/yellow` and install it using the browser, selecting english as default language: `git clone https://github.com/datenstrom/yellow.git test`
2. Install polish language from command line and set it as default: `php yellow.php install polish`
3. Log into the webmaster's user account and change language in the settings.
4. Remove all unneeded languages from command line: `php yellow.php uninstall english german swedish`

Website still without errors, and the language selection in user settings is gone. I however can't select polish during installation as default language, this probably only would work if my browser's language is polish.

**Reply by kanbeq** [reply]
1. Downloaded and installed by choosing English
2. I started the installation of the Polish language. I got a message that I already have this extension/language
3. I changed the user settings to Polish
4. I uninstalled English and German.
During this process, I received the following notices:
```
PHP Notice: Only variables should be passed by reference in ...\yellow\system\extensions\english.php on line 19

Notice: Only variables should be passed by reference in ...\yellow\system\extensions\english.php on line 19
PHP Notice: Only variables should be passed by reference in ...\yellow\system\extensions\german.php on line 19

Notice: Only variables should be passed by reference in ...\yellow\system\extensions\german.php on line 19
```
5. When I launched the page in the browser, an error popped up saying that I don't have the 'en' language
6. I changed the language in files yellow-system.ini and yellow-language.ini to 'pl'
7. The error is gone. No language is shown in the settings

You could say that the error is eliminated. But to fix it on a previous install, I would have to reinstall yellow

**Reply by schulle4u** [reply]
Yes, I think this behaviour can be expected when Removing something that is still in use by user or system configuration. First you should change **all** language configuration to your desired language, then remove the unneeded extensions. Actually this should also work in existing installations. 

Just for curiosity, why do you wanna get rid of these 20 or 30 KB text files?

**Reply by kanbeq** [reply]
I was more concerned with the error itself.
Somehow these files do not bother me much. Since I have the option of removing them, I do so because I do not need them. 😉

To sum up - I have to change all dependencies related to extensions first, and then remove them. So the problem is solved 🙂

**Reply by kanbeq** [reply]
Unfortunately, the error persists. I forgot to mention that I have extensions installed. The error calls the extension "[private](https://github.com/schulle4u/yellow-extensions-schulle4u/tree/master/private)". When I delete them, the error disappears.

**Reply by schulle4u** [reply]
Thanks for the hint, I'll investigate this. The private extension uses a single language file for all supported languages, maybe there is a problem if the strings don't exist for a certain language. I'll check after barbecue. 😀

**Reply by markseuffert** [reply]
I can confirm the bug, I found a similar one a while ago but haven't had time to upload my bug fix.

**Reply by kanbeq** [reply]
> I have to change all dependencies related to extensions

As I thought a little longer, I had the idea of removing all unused languages from the text file of this extension. And it helped. But I think you have a better solution.

**Comment by markseuffert** [comment, marked as answer]
I updated the [core extension](https://github.com/annaesvensson/yellow-core). This error was caused by incomplete/unused languages. In theory languages should have been automatically cleaned  up during loading, but ArraryObject has an unexpected behaviour when being modified in a foreach loop. The iterator skips the next element when the current element is unset in a foreach loop. This has already caused us problems in the past, must have slipped trough testing again. See also [PHP bug 76387](https://bugs.php.net/bug.php?id=76387).

Should be fixed now, please try it out.

**Reply by kanbeq** [reply]
All right now, thanks for all the help
